### ISSUE TITLE
Adding an invariant to a contract function is silently ignored when arming the smart contract

### ISSUE THREAD

**Issue by Magicking** [ConsenSys/scribble#6]
Example for the batch deposit contract where an invariant is placed before a function* https://gist.github.com/Magicking/fe63df8e8fce9a0710c53bc34fec7ac9/0890379fc80c3f55aba53dc52639723c48be6dfc#file-batchdeposit-sol-L72
And the result from `npx scribble --arm batchDeposit.sol` produced this armed smart contract https://gist.github.com/Magicking/fe63df8e8fce9a0710c53bc34fec7ac9/e57c7b5b094083dfea77bfcd3434528e6440d948#file-batchdepositarmed-sol-L64 where the NatSpec documentation is remove (which I think may be a separate bug) and the assert for this invariant is silently ignored.

*The solution found was to return the value and verify it's value, see here https://gist.github.com/Magicking/fe63df8e8fce9a0710c53bc34fec7ac9

**Comment by cd1m0** [comment]
Thanks for filing! So if I understand correctly, the issue is that this invariant is not being emitted:

```
    /// invariant {:msg "supplied ether value mismatch the total deposited sum"} deposited_amount == msg.value;
    function batchDepositVariable(bytes[] calldata pubkeys, bytes[] calldata withdrawal_credentials, bytes[] calldata signatures, bytes32[] calldata deposit_data_roots, uint64[] calldata amounts) external payable {
        uint256 deposited_amount = 0;
    ....
```

Right? 

So if this is the issue, then the fundamental problem is that scribble should have thrown an error, because `invariant`s are only allowed on the contract level, not the functions level. Will use this issue to track adding that fix. Let me know if I have correctly understood your problem.

(Related to the missing docstring on the instrumented code - that is the expected current behavior. We decide to remove the docstrings in the instrumented code to prevent people from accidentally re-instrumenting already instrumented code).

**Comment by Magicking** [comment]
Forgot to mention that indeed the issue raised here is about the invariant not throwing an error when applied to functions.

Regarding the NatSpect documentation missing, the problem is that when you upload the armed version to a website like etherscan, the parameters documentations is lost!

**Comment by cd1m0** [comment]
Originally, our intention was that only the annotated code would be presented to outside users and uploaded to etherscan, and that instrumented code would only be used internally by mythril, MythX and local tests. If you have a use-case where you want to present the instrumented code to the outside world, we can definitely change instrumentation to preserve docstrings, or add an option for including docstrings.

**Comment by cd1m0** [comment]
Fixed in 0.3.3
